dlib/cmake_utils/test_for_neon/neon_test.cpp

```diff
@@ -0,0 +1,26 @@
+#ifdef __ARM_NEON__
+#warning "__ARM_NEON__ is defined."
+int foo(int a, int b)
```

Comment:
This code seems too complex.  Wouldn't it work if all that was in here was this?

```
#ifndef __ARM_NEON__
#error "No NEON"
#endif
int main() {}
```